internal/command/cloud.go

```diff
@@ -0,0 +1,102 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/terraform/internal/cloudplugin"
+	"github.com/hashicorp/terraform/internal/cloudplugin/cloudplugin1"
+	"github.com/hashicorp/terraform/internal/logging"
+)
+
+// CloudCommand is a Command implementation that interacts with Terraform
+// Cloud for operations that are inherently planless. It delegates
+// all execution to an internal plugin.
+type CloudCommand struct {
+	Meta
+}
+
+const (
+	// DefaultCloudPluginVersion is the implied protocol version, though all
+	// historical versions are defined explicitly.
+	DefaultCloudPluginVersion = 1
+
+	// ExitRPCError is the exit code that is returned if an plugin
+	// communication error occurred.
+	ExitRPCError = 99
+)
+
+var (
+	// Handshake is used to verify that the plugin is the appropriate plugin for
+	// the client. This is not a security verification.
+	Handshake = plugin.HandshakeConfig{
+		MagicCookieKey:   "TF_CLOUDPLUGIN_MAGIC_COOKIE",
+		MagicCookieValue: "721fca41431b780ff3ad2623838faaa178d74c65e1cfdfe19537c31656496bf9f82d6c6707f71d81c8eed0db9043f79e56ab4582d013bc08ead14f57961461dc",
+		ProtocolVersion:  DefaultCloudPluginVersion,
+	}
+)
+
+func (c *CloudCommand) proxy(args []string, stdout, stderr io.Writer) int {
+	client := plugin.NewClient(&plugin.ClientConfig{
+		HandshakeConfig:  Handshake,
+		AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
+		Cmd:              exec.Command("./terraform-cloudplugin"),
+		Logger:           logging.NewCloudLogger(),
+		VersionedPlugins: map[int]plugin.PluginSet{
+			1: {
+				"cloud": &cloudplugin1.GRPCCloudPlugin{},
+			},
+		},
+	})
+	defer client.Kill()
+
+	// Connect via RPC
+	rpcClient, err := client.Client()
+	if err != nil {
+		fmt.Fprintf(stderr, "Failed to create cloud plugin client: %s", err)
+		return ExitRPCError
+	}
+
+	// Request the plugin
+	raw, err := rpcClient.Dispense("cloud")
+	if err != nil {
+		fmt.Fprintf(stderr, "Failed to request cloud plugin interface: %s", err)
+		return ExitRPCError
+	}
+
+	// Proxy the request
+	cloud1 := raw.(cloudplugin.Cloud1)
+	return cloud1.Execute(args, stdout, stderr)
+}
+
+// Run runs the cloud command with the given arguments.
+func (c *CloudCommand) Run(args []string) int {
+	args = c.Meta.process(args)
+
+	// TODO: Download and verify the signing of the terraform-cloudplugin
+	// release that is appropriate for this OS/Arch
+	if _, err := os.Stat("./terraform-cloudplugin"); err != nil {
+		c.Ui.Warn("terraform-cloudplugin not found. This plugin does not have an official release, yet.")
+		return 1
+	}
+
+	// TODO: Need to use some type of c.Meta handle here
+	return c.proxy(args, os.Stdout, os.Stderr)
+}
+
+// Help returns help text for the cloud command.
+func (c *CloudCommand) Help() string {
```

Comment:
`terraform cloud non-existing-subcommand` is saying that you can run `terraform cloud -help` to see all cloud subcommands, but if the implementation for Help is not complete, then maybe let's not advertise this option. Alternatively you can recommend to run `terraform cloud` instead, which shows available commands.